smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftSymbolProvider.kt

```diff
@@ -284,6 +297,88 @@ class SwiftSymbolProvider(private val model: Model, val swiftSettings: SwiftSett
         return ModelFileUtils.filename(swiftSettings, name)
     }
 
+    /**
+     * Resolve default value for a given shape and save it as a property in symbol builder if needed.
+     *
+     * The default trait can be applied to list shape, map shape, and all simple types as per Smithy spec.
+     * Both the member shape and the target shape may have the default trait.
+     *
+     * There exist default value restrictions for the following shapes:
+     *  - enum: can be set to any valid string value of the enum.
+     *  - intEnum: can be set to any valid integer value of the enum.
+     *  - document: can be set to null, true, false, string, numbers, an empty list, or an empty map.
+     *  - list: can only be set to an empty list.
+     *  - map: can only be set to an empty map.
+     */
+    private fun handleDefaultValue(shape: Shape, builder: Symbol.Builder): Symbol.Builder {
+        // Skip if the current shape does not have a default trait or if it's a member shape with @clientOptional trait
+        if (!shape.hasTrait<DefaultTrait>() || shape.hasTrait<ClientOptionalTrait>()) {
+            return builder
+        }
+        // Retrieve literal default value as a string from default trait
+        val defaultValueLiteral = shape.getTrait<DefaultTrait>()!!.toNode().toString()
+        // If default value is "null", it is explicit notation for no default value. Return unmodified builder.
+        if (defaultValueLiteral.equals("null")) { return builder }
+
+        // The current shape may be a member shape or a root level shape.
+        val targetShape = when (shape) {
+            is MemberShape -> {
+                // If containing shape is an input shape, return unmodified builder.
+                if (model.expectShape(shape.container).hasTrait<InputTrait>()) { return builder }
+                model.expectShape(shape.target)
+            }
+            else -> { shape }
+        }
```

Comment:
this looks like a good candidate to move to a helper function to improve readability